docs: Doc update part 1 (#266)
samba-in-kubernetes/samba-operator: docs/README.md

# Samba Operator Documentation

## Introduction

The samba-operator is a Kubernetes operator designed to export other storage

> layers as SMB shares. These SMB shares can be accessed from within a Kubernetes

Reviewer: ```suggestion
layers as SMB shares. These SMB shares can be accessed from within theKubernetes cluster hosting the samba operator
```

Author: done

Reviewer: thanks!

> cluster or outside the Kubernetes cluster.  The shares may be enabled for

Reviewer: ```suggestion
cluster or from outside the Kubernetes cluster.  The shares may be enabled for
```

Author: done

Active Directory authentication.  Multiple shares can be served by a single
server instance to reduce resource consumption.  An experimental feature allows
the shares to be served by a CTDB enabled cluster of Samba servers.

The samba-operator is designed around the idea that what people want are shares
and the data within them; and that servers are an implementation detail.  With

> this in mind, the samba-operator's primary resource is the SmbShare.  When

Reviewer: ```suggestion
this in mind, the samba-operator's primary resource is the [SmbShare](./resources/SmbShare.md).  When
```

Author: done

> moving beyond the most basic cases, it provides the SmbSecurityConfig resource

Reviewer: ```suggestion
moving beyond the most basic cases, it provides the [SmbSecurityConfig](./resources/SmbSecurityConfig.md) resource
```

Author: done

Reviewer: Thanks!

for configuring Active Directory integration or specifying the users/groups for

> authentication and access. The SmbCommonConfig resource is used to define

Reviewer: ```suggestion
authentication and access. The [SmbCommonConfig](./resources/SmbCommonConfig.md) resource is used to define
```

Author: done

> authentication and access. The SmbCommonConfig resource is used to define

Reviewer: ```suggestion
authentication and access. The [SmbCommonConfig](./resources/SmbCommonConfig.md) resource is used to define
```

Author: done

shared networking and Kubernetes cluster integration properties. Both

> SmbSecurityConfig and SmbCommonConfig can be referenced by multiple SmbShare

Reviewer: ```suggestion
[SmbSecurityConfig](./resources/SmbSecurityConfig.md) and [SmbCommonConfig](./resources/SmbCommonConfig.md) can be referenced by multiple [SmbShare](./resources/SmbShare.md)
```

Author: done

# Samba Operator Documentation

## Introduction

The samba-operator is a Kubernetes operator designed to export other storage
layers as SMB shares. These SMB shares can be accessed from within the
Kubernetes cluster hosting the samba-operator or from outside the Kubernetes
cluster.  The shares may be enabled for Active Directory authentication.
Multiple shares can be served by a single server instance to reduce resource
consumption.  An experimental feature allows the shares to be served by a CTDB
enabled cluster of Samba servers.

> The samba-operator is designed around the idea that what people want are shares

Author: done

> The samba-operator is designed around the idea that what people want are shares

Reviewer: thanks!

> and the data within them; and that servers are an implementation detail.  With

Reviewer: ```suggestion
and the data within them; and that servers are an implementation detail irrelevant for most users.  With
```

Author: done
